### PR TITLE
WT-13048 Use a typedef for WT_REF::state

### PR DESCRIPTION
**Pull request by marcbutler**
Implements the type alias WT_REF_STATE for WT_REF::state field.

### CODE REVIEW

File: src/btree/bt_compact.c [hunk 1]
```diff
@@ -162,8 +162,8 @@ __compact_page(WT_SESSION_IMPL *session, WT_REF *ref, bool *skipp)
     WT_ADDR_COPY copy;
     WT_BM *bm;
     WT_DECL_RET;
+    WT_REF_STATE_TYPE previous_state;
```
Comment (ajmorton) [inline src/btree/bt_compact.c:165]: Suggestion: Shorten the name to `WT_REF_STATE`

**Review by ajmorton — APPROVED** [review]
LGTM 👍 
One naming suggestion to take or leave.

File: src/btree/bt_compact.c (hunk 1 above)
Comment (marcbutler) [reply src/btree/bt_compact.c:165]: Happy to shorten, took the prompt form the `typedef enum` aliases which use this convention. Any other reviewers care to weigh in?

Comment (jeremythorp) [reply src/btree/bt_compact.c:165]: I support Andrew's suggestion to shorten the name, but it's not a must have change for me.

**Review by jeremythorp — APPROVED** [review]
LGTM!

File: src/btree/bt_compact.c (hunk 1 above)
Comment (marcbutler) [reply src/btree/bt_compact.c:165]: @agorrod Interested in your take on this.

Comment (korteland) [reply src/btree/bt_compact.c:165]: I would prefer to shorten it. Most of the `_TYPE` suffixes describe the type of something (e.g. `WT_BTREE_TYPE`). There are plenty of typedefs without the suffix, e.g. `WT_VERBOSE_CATEGORY`.

File: src/include/btmem.h [hunk 2]
```diff
@@ -805,6 +805,19 @@ struct __wt_page_walk_skip_stats {
     size_t total_inmem_del_pages_skipped;
 };
 
+/*
+ * Type used by WT_REF::state and valid values.
+ *
+ * Declared well before __wt_ref struct as the type used in other structures in this header.
```
Comment (korteland) [inline src/include/btmem.h:811]: Nitpick: `type used` -> `type is used`

**Review by korteland — COMMENTED** [review]
Some smaller comments, but I'll also use this review to say that I'm comfortable with the larger goal (discussed offline) of Futex support.

File: src/btree/bt_compact.c (hunk 1 above)
Comment (marcbutler) [reply src/btree/bt_compact.c:165]: The shortens have it!

File: src/include/btmem.h (hunk 2 above)
Comment (marcbutler) [reply src/include/btmem.h:811]: thanks!

**Comment by marcbutler** [comment]
> Some smaller comments, but I'll also use this review to say that I'm comfortable with the larger goal (discussed offline) of Futex support.

I'm glad you found that useful. Irrespective of that goal I think this change is useful unto itself. As the value semantic is captured in fields in other structures (ie `WT_PAGE_DELETED`) in addition to being captured in variables. This change makes that relationship explicit independent of the variable naming.

**Review by korteland — APPROVED** [review]
LGTM!

**Pull request merged**